libcalico-go/lib/ipam/ipam_block_reader_writer_test.go

```diff
@@ -384,8 +384,8 @@ var _ = testutils.E2eDatastoreDescribe("IPAM affine block allocation tests", tes
 						defer wg.Done()
 
 						testhost := "same-host"
-						affinityCfg := AffinityConfig{
```

Comment:
We should add tests here and/or in ipam_test.go that check the edge cases related to virtual allocations.

e.g.,

- Allocate IPs to a host named `virtual-loadbalancer`, allocate actual virtual load balancer IPs.
- Release affinities in various orders, make sure only the desired affinities are released.
- Once affinities are released, also delete IPs from the blocks - make sure state is correct for each.

Reply:
We should also add tests here that simply check reading / writing virtual affinities and blocks works as expected - i.e., write a block, read it back, compare the two and make sure they match exactly.